### ISSUE TITLE
[Suggestion] Bonus roll effects could use a couple more flags

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#420]
In GitLab by @Khozma on Jul 6, 2021, 19:41

For the new "bonus effects applied after the roll but before the roll is applied" system, there's some functionality that is missing with the new flags that could be useful to support:

1. Right now, there doesn't appear to be a way to add a bonus such as a Bardic Inspiration to a skill check - it only applies to an ability check. I think flags.midi-qol.optional.Name.check should either apply to skill checks as well, since they're also considered ability checks and I'm not sure there's any feature that would need exclusive functionality here, or there should also be a flags.midi-qol.optional.Name.skill flag.

2. There's currently no flag to add bonus dice to weapon damage. The College of Valour bard comes to mind here, but I'm sure there's other features this would be useful for, as well. This one is pretty easy to do manually with auto damage off, but it'd be nice to have, especially so players don't forget about the option!

3. This one is a little more niche, but it would be useful to have a flag that lets you add a bonus dice to your AC as a reaction against an attack that would hit you. The Valour bard's bardic inspiration dice can be added to AC using a reaction this way. This would also be helpful for a similar College of Swords feature, Defensive Flourish.

**Comment by tposney** [comment]
Skill is easy enough to add and probably prefer that than having ability check auto apply to skill.
Damage also makes sense.
Ac should be possible, but I'll have to have a tinker with that.

**Comment by tposney** [comment]
Skills have been added as an extra flag and will go out in 0.8.31.  
Damage/AC need more thought. Damage because it occurs at a different point in the roll process and I have to work out how to have it apply just once without creating an effect.  
AC is even trickier, or maybe not, it's a bit like a reaction which already let's you add AC. That might be the solution add an effect to boost AC that lasts 1 turn/1hit.

**Comment by tposney** [comment]
Turns out I found a way to do this, so both 
flags.midi-qol.optional.NAME.ac and  
flags.midi-qol.optional.NAME.damage  
are supported in 0.8.33

**Comment by tposney** [comment]
In GitLab by @Khozma on Jul 13, 2021, 03:35

Many thanks, you work fast!

**Comment by tposney** [comment]
In GitLab by @Khozma on Jul 13, 2021, 04:26

Found a couple minor issues on the new flags after trying them:

1. For the damage flag, the dice aren't currently rolled twice on a crit.

2. For the AC flag, as far as I can tell (might be missing something), there's currently no way to see the roll on the dice after you use it - works perfectly, just the result of the roll is invisible.

**Comment by tposney** [comment]
For 1, yeah the dice are not rolled twice on critical. The cases I looked at suggest that it should not be rolled twice - the other option is a damage bonus macro, which can do the critical damage part.  

For 2, you're right. I could not come up with a good way to manage the display - one requirement is that once the roll is done the roll should continue, but the idea that you see what was rolled also makes sense - I'll have to think about that.

**Comment by tposney** [comment]
The ac notification to chat should be working now.

**Comment by tposney** [comment]
There's now an option to display the before and after roll for optional effects so that should cover all the cases.
